java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/WindowAggPlan.scala

```diff
@@ -149,50 +148,38 @@ object WindowAggPlan {
     }
 
     // Get repartition keys and orderby key
-    // TODO: Now it does not support columns with the same names because it uses column name for repartition
-    val repartitionColNames = PhysicalNodeUtil.getRepartitionColumnNames(windowAggNode, inputDf).asJava
+    // TODO: Support multiple repartition keys and orderby keys
     val repartitionColIndexes = PhysicalNodeUtil.getRepartitionColumnIndexes(windowAggNode, inputDf)
-    val orderbyColName = PhysicalNodeUtil.getOrderbyColumnName(windowAggNode, inputDf)
-    val orderbyColIndex = PhysicalNodeUtil.getOrderbyColumnIndex(windowAggNode, inputDf)
-    val orderbyColType = inputDf.schema(orderbyColIndex).dataType
+    val percentileColIndex = PhysicalNodeUtil.getOrderbyColumnIndex(windowAggNode, inputDf)
+    val percentileColType = inputDf.schema(percentileColIndex).dataType
 
     // Register the input table
-    val inputTableName = "_INPUT_TABLE_" + uniqueNamePostfix
-    inputDf.createOrReplaceTempView(inputTableName)
-
-    val partColumnName = "_PART_" + ctx.getConf.windowSkewOptPostfix
-    val expandColumnName = "_EXPAND_" + ctx.getConf.windowSkewOptPostfix
-    val addColumnsTableName = "_ADD_COLUMNS_TABLE_" + uniqueNamePostfix
+    val partColName = "_PART_" + uniqueNamePostfix
+    val expandColName = "_EXPAND_" + uniqueNamePostfix
 
     val quantile = math.pow(2, ctx.getConf.skewLevel.toDouble)
-    val schemas = scala.collection.JavaConverters.seqAsJavaList(inputDf.schema.fieldNames)
-    var keyScala = repartitionColNames.asScala
 
-    val addColumnsDf = if (ctx.getConf.windowSkewOptConfig.equals("")) { // Do not use skew config
+    val addColumnsDf = if (ctx.getConf.windowSkewOptConfig.equals("")) {
+      // Do not use skew config
       // 1. Analyze the data distribution
-      val distributionTableName = "_DISTRIBUTION_TABLE_" + uniqueNamePostfix
-      val countColumnName = "_COUNT_" + uniqueNamePostfix
+      val partitionColName = "_PARTITION_" + uniqueNamePostfix
 
-      val distributionSqlText = SkewUtils
-        .genPercentileSql(inputTableName, quantile.intValue(), repartitionColNames, orderbyColName, countColumnName)
-      logger.info(s"Generate distribution sql: $distributionSqlText")
-      val distributionDf = ctx.sparksql(distributionSqlText)
-      distributionDf.createOrReplaceTempView(distributionTableName)
+      val distributionDf = genDistributionDf(inputDf, quantile.intValue(), repartitionColIndexes,
```

Comment:
It would be better to call by `SkewDataFrameUtils.genDistributionDf` since it is not defined in this file.